evap/staff/tests/test_views.py

```diff
@@ -1136,11 +1138,20 @@ def helper_semester_state_views(self, evaluation, old_state, new_state):
     def test_semester_publish(self):
         participant1 = baker.make(UserProfile, email="foo@example.com")
         participant2 = baker.make(UserProfile, email="bar@example.com")
-        evaluation = baker.make(Evaluation, course=self.course, state='reviewed',
-                                participants=[participant1, participant2], voters=[participant1, participant2])
+        evaluation = baker.make(
+            Evaluation,
+            course=self.course,
+            state='reviewed',
+            participants=[participant1, participant2],
+            voters=[participant1, participant2]
+        )
 
         self.helper_semester_state_views(evaluation, "reviewed", "published")
-        self.assertEqual(len(mail.outbox), 2)
+        self.assertEqual(len(mail.outbox), 3)
+        self.assertEqual(
+            {recipient for msg in mail.outbox for recipient in msg.to},
+            {participant1.email, participant2.email, self.responsible.email}
+        )
```

Comment:
This doesn’t keep the bag-of-lists-semantic – we could accidentally send an email twice to an user. Is this intended?

Reply:
Nope, wasn't intended, good catch